google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/UserSpecifiedPathDeploymentSource.java

```diff
@@ -59,13 +60,19 @@ public Icon getIcon() {
   @NotNull
   @Override
   public String getPresentableName() {
-    return name;
+    return String.format("[%s] ", DateTime.now().toString("yyyy-MM-dd HH:mm:ss")) + name;
```

Comment:
Have you given any thought to extracting this method to an abstract class or so? This method and setName() seem to be the same throughout. I don't think this alone warrants creating an abstract class, just want to know what you think.